src/Agent.php

```diff
@@ -61,18 +65,28 @@ public function getCollector(string $name): DataCollector
         return $this->collectors->get($name);
     }
 
+    public function setCurrentTransaction(Transaction $transaction): void
```

Comment:
Do we need to unset the transaction once is sent? Thinking about async jobs here.

Reply:
Yes. I added this to the `Agent::send()` method which we had previous overridden anyway. That seems to be the best place to hook this in.